src/CodeGen/ODataT4CodeGenerator.cs

```diff
@@ -425,6 +435,123 @@ public void ValidateAndSetIgnoreUnexpectedElementsAndAttributesFromString(string
     this.IgnoreUnexpectedElementsAndAttributes = boolValue;
 }
 
+public void ValidateAndSetCustomHttpHeadersFromString(string stringValue) 
+{
+    this.CustomHttpHeaders = new Dictionary <string, string> ();
+
+    if (String.IsNullOrWhiteSpace(stringValue)) {
+        return;
+    }
+
+    if (ValidateParantheses(stringValue) && ValidateFirstLast(stringValue) && ValidateDelimeters(stringValue)) 
+    {
+        string header = stringValue.TrimEnd('}', ' ');
+        string[] elements = header.Split('}');
+        foreach (var element in elements)
+        {
+            string _element = element.TrimStart(',', '{', ' ');
+            string[] array = _element.Split(':');
+            CustomHttpHeaders.Add(array[0], array[1]);
+        }
+    }
+    else 
+    {
+        throw new ArgumentException("Please check your custom http headers string format! You have set the custom http header but not use the format '{headerName:value},{headerName:value}'");

```

Comment:
does code gen has the central error message management?

Reply:
No it doesn't

Reply:
So, it's not allowed to "{Acce}:pt:application/json:abc}" ?

---
In reply to: [368422032](https://github.com/OData/odata.net/pull/1637#discussion_r368422032) [](ancestors = 368422032)